text/002-css-rules.md

```diff
@@ -1,894 +1,991 @@
+
+**Usage:**
+
+```typescript
+button({
+  color: "accent",
+  size: "large",
+  rounded: true
+});
+```
+
+**Compiled:**
+
+```css
+.[FILE_NAME]_button__[HASH] {
+  color: black;
+  background-color: white;
+  border-radius: 6px;
+}
+
+.[FILE_NAME]_button_color_brand__[HASH] {
+  color: #ffffa0;
+  background-color: blueviolet;
+}
+.[FILE_NAME]_button_color_accent__[HASH] {
+  color: #ffe4b5;
+  background-color: slateblue;
+}
+
+.[FILE_NAME]_button_size_small__[HASH] {
+  padding: 12px;
+}
+.[FILE_NAME]_button_size_medium__[HASH] {
+  padding: 16px;
+}
+.[FILE_NAME]_button_size_large__[HASH] {
+  padding: 24px;
+}
+```
+
+## 5. Toggle Variants
+
+[Stitches's `boolean variants`](https://stitches.dev/docs/variants#boolean-variants) are a special case, but the syntax for defining them is awkward.
+
+Therefore, we introduce a specialized syntax.
+
+**Code Before:**
+
+```typescript
+const button = rules({
+  // base styles
+
+  variants: {
+    // common variants
+    rounded: {
+      true: { borderRadius: 999 }
+    }
+  }
+});
+```
+
+**Code After:**
+
+```typescript
+const button = rules({
+  // base styles
+
+  toggles: {
+    rounded: { borderRadius: 999 }
+  }
+
+  variants: {
+    // common variants
+  }
+});
+```
+
+It seems also awkward to use `true` even when using.
+
+It can be represented through an array.  
+This is also a representation isomorphism with `css()`.
+
+**Usage Before:**
+
+```typescript
+button({
+  rounded: true,
+  color: "accent",
+  size: "large"
+});
+```
+
+**Usage After:**
+
+```typescript
+button([
+  "rounded",
+  {
+    color: "accent",
+    size: "large"
+  }
+]);
+```
+
+## 6. Compound Variants
+
+[Stitche's `Compound Variants`](https://stitches.dev/docs/variants#compound-variants) is an effective way to set up additional css by leveraging the combination of variations you have already set up.
+
+However, the method of writing the conditions seems quite inconvenient.  
+So we want to improve the UX in this area.
+
+**Code:**
+
+```typescript
+const button = rules({
+  // base style
+
+  variants: {
+    color: {
+      brand: {
+        color: "#FFFFA0"
+      },
+      accent: {
+        color: "#FFE4B5"
+      }
+    },
+    size: {
+      small: { padding: 12 },
+      medium: { padding: 16 },
+      large: { padding: 24 }
+    },
+    border: {
+      basic: "1px solid #000"
+    }
+  },
+  // case 1
+  compoundVariants: [
+    {
+      condition: [color.brand, size.small],
+      style: {
+        fontSize: "16px"
+      }
+    },
+    {
+      condition: [color.accent, size.large],
+      style: {
+        fontSize: "24px",
+        fontWeight: "bold"
+      }
+    }
+  ],
+  // case 2
+  compoundVariants: {
+    basic: {
+      variants: [color.brand, size.small],
+      style: {
+        fontSize: "16px"
+      }
+    },
+    title: {
+      variants: [color.accent, size.large],
+      style: {
+        fontSize: "24px",
+        fontWeight: "bold"
+      }
+    }
+  }
+});

```

Comment:
After I made this, I found that it's similar with [Vanilla Extract's styleVariants](https://vanilla-extract.style/documentation/api/style-variants/).
So depends on our discussion, It will separated from this to make another independent feature.